### PR TITLE
[1.x] Validate new owner belongs to team before transferring.

### PR DESCRIPTION
**Pull request by joelbutcher**
This PR introduces a fix for a possible bug whereby teams could potentially be transferred to users that aren't a part of that team, as pointed out by @iantasker in #4.

A new check is made on the form submit action and a validation step is added to the `TransferTeam` logic.

### CODE REVIEW

File: src/Http/Livewire/TeamTransferForm.php [hunk 1]
```diff
@@ -83,15 +86,25 @@ public function transferTeam(TransferTeam $transferrer)
             ]);
         }
 
+        $teamMember = Jetstream::findUserByEmailOrFail($this->transferTeamForm['email']);
+
+        if ($this->team->hasUser($teamMember)) {
```
Comment (iantasker) [inline src/Http/Livewire/TeamTransferForm.php]: You need to invert the logic here as you only want to display a warning if the user isn't a member of the team. 

```suggestion
        if (!$this->team->hasUser($teamMember)) {
```

**Review by iantasker — CHANGES_REQUESTED** [review]

File: src/Http/Livewire/TeamTransferForm.php [hunk 2]
```diff
@@ -83,15 +86,25 @@ public function transferTeam(TransferTeam $transferrer)
...
+            $this->dangerBanner('You cannot transfer team ownership to users outside of this team.');
+
+            $this->confirmingTransferTeam = false;
```
Comment (iantasker) [inline src/Http/Livewire/TeamTransferForm.php]: If the user is not part of the team the modal doesn't currently close.
 
```suggestion
            $this->confirmingTeamTransfer= false;
```

File: src/Http/Livewire/TeamTransferForm.php [hunk 3]
```diff
@@ -83,15 +86,25 @@ public function transferTeam(TransferTeam $transferrer)
             ]);
         }
 
+        $teamMember = Jetstream::findUserByEmailOrFail($this->transferTeamForm['email']);
+
+        if ($this->team->hasUser($teamMember)) {
+            $this->dangerBanner('You cannot transfer team ownership to users outside of this team.');
```
Comment (iantasker) [inline src/Http/Livewire/TeamTransferForm.php]: Updated string to read a little better.
 
```suggestion
            $this->dangerBanner('You cannot transfer team ownership to a user outside of this team.');
```

**Review by iantasker — CHANGES_REQUESTED** [review]

File: src/Http/Livewire/TeamTransferForm.php [hunk 4]
```diff
@@ -83,15 +86,25 @@ public function transferTeam(TransferTeam $transferrer)
             ]);
         }
 
+        $teamMember = Jetstream::findUserByEmailOrFail($this->transferTeamForm['email']);
+
+        if (!$this->team->hasUser($teamMember)) {
```
Comment (joelbutcher) [inline src/Http/Livewire/TeamTransferForm.php]:
```suggestion
        if (!$this->team->hasUser($teamMember)) {
        if (! $this->team->hasUser($teamMember)) {
```

**Pull request merged**